internal/seed/probes.go

```diff
@@ -0,0 +1,164 @@
+package seed
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/akash-network/rpc-proxy/internal/block"
+	rpchttp "github.com/cometbft/cometbft/rpc/client/http"
+	"github.com/cosmos/cosmos-sdk/client/grpc/cmtservice"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"io"
+	"net/http"
+	"time"
+)
+
+// Probe is the interface that wraps the Probe method.
+//
+// Probe probes the given node for health.
+// A healthy node must return a healthy status.
+// Health is subject to the caller's interpretation of the returned Status.
+type Probe interface {
+	Probe(node Node) (Status, error)
+}
+
+// ProbeFunc type is an adapter to allow the use of ordinary functions as probes.
+// If a given function f is a function with the appropriate signature, ProbeFunc(f) is a Probe that calls f.
+type ProbeFunc func(node Node) (Status, error)
+
+// Probe implements the Probe interface for ProbeFunc to allow for defining probes as standalone function
+func (f ProbeFunc) Probe(node Node) (Status, error) {
+	return f(node)
+}
+
+// RPCProbe probes an RPC Node.
+// It queries the Node status through RPC and tries to set the latest block height globally.
+func RPCProbe(node Node) (Status, error) {
+	client, err := rpchttp.New(node.Address, "/") // TODO: Test this
```

Comment:
is there an issue to write the test or will it be included in this PR?

Reply:
This `TODO` should have been removed